lib/core/request.js

```diff
@@ -164,10 +164,21 @@ class Request {
         processHeader(this, headers[i], headers[i + 1])
       }
     } else if (headers && typeof headers === 'object') {
-      const keys = Object.keys(headers)
-      for (let i = 0; i < keys.length; i++) {
-        const key = keys[i]
-        processHeader(this, key, headers[key])
+      if (headers[Symbol.iterator]) {
+        for (const header of headers) {
+          if (!Array.isArray(header) || header.length !== 2) {
+            throw new InvalidArgumentError('headers must be in key-value pair format')
+          }
+          const key = header[0]
+          const value = header[1]
```

Comment:
I wonder if we can use it this way
```suggestion
   const [key, value] = header;
```

Reply:
Long story short, this approach is absolutely fine.
For a long time I've been thaught that your suggestion is slower due to internal iterator creation.
Now I double checked and find out that v8 devs already fixed it.
Everything is good 👍

Reply:
It's great 🚀